File: Posts/Data-Strucutres-And-Algorithms/Recursion/Problem-Four.py
def backtracking(n: int, index: int, x: list):
	if index == n:
		print("".join(map(str, x)))
	else:
		for i in range(0, 2):
			x[index] = i
			backtracking(n, index + 1, x)


n = int(input())
x = [0 for i in range(0, n)]
backtracking(n, 0, x)

File: posts/study/data-strucutres-and-algorithms/recursion/Problem-Four.cpp
#include <bits/stdc++.h>

using namespace std;

void backtracking(const int n, int index, int*& x) {
	if (index == n) {
		for (int i = 0; i < n; ++i) {
			cout << x[i];
		}
		cout << "\n";
		return;
	}
	for (int i = 0; i <= 1; ++i) {
		x[index] = i;
		backtracking(n, index + 1, x);
	}
}

int main() {
	int n;
	cin >> n;
	int* x = new int [n];
	backtracking(n, 0, x);
	delete[] x;
	return 0;
}
